citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -387,6 +391,9 @@ public CitationResults getGroupItems(IUser user, String groupId, String collecti
             }
         } else {
             group = groupOptional.get();
+            if (!group.getUsers().contains(user.getUsername())){
+                group.getUsers().add(user.getUsername());
+            }
```

Comment:
this should not be needed, should it? this else should only be called if the group is present and the user has access to it?